elements/pf-alert/pf-alert.ts

```diff
@@ -0,0 +1,317 @@
+    const inDemo = this.closest('.demo-with-arrows') !== null;
+    const hasDescription = this.querySelector('p') !== null;
+    const showArrow = inDemo;
+    const arrowDirection = hasDescription ? 'angle-down' : 'angle-right';
+    const footer = html`<footer class="${classMap({ hasActions })}"
+            <!-- Provide actions that the user can take for the alert -->
+            <slot name="actions"></slot>
+          </footer>`;
+    return html`
+      <section id="container"
+              part="container"
+              class=
+              ${classMap({
+      hasBody,
+      light: true,
+      [state]: true,
+      [variant]: !!variant,
+    })}
+             role="alert"
+             aria-hidden="false"
+             color-palette="lightest">
+        <div id="left-column" style="display: flex; align-items: center; gap: 0.5rem;">
+          ${showArrow ? html`
+          <pf-icon 
+            id="arrow-icon"
+            set="fas"
+            icon="${arrowDirection}"
+            class="alerts-page-only"
+            style="--pf-c-icon--Color: var(--_icon-color); font-size: 16px; height: 16px; width: 16px;">
+          </pf-icon>` : ''}
+          <pf-icon 
+            id="icon"
+            set="fas" 
+            icon="${this.#icon}"  
+            style="--pf-c-icon--Color: var(--_icon-color); font-size: 36px; height: 36px; width: 36px;">
+          </pf-icon>
+        </div>
+        <div id="middle-column">
+          <header ?hidden="${!_isServer && this.#slots.isEmpty('header')}">
+            <div id="header">
+              <!-- Provide a header for the alert message. -->
+              <slot name="header"></slot>
+            </div>${!this.dismissable && this.variant !== 'toast' ? '' : html`
+            <div id="header-actions">
+                <pf-icon
+                  id="close-button"
+                  set="fas"
+                  icon="times"
+                  role="button"
+                  tabindex="0"
+                  aria-label="Close"
+                </pf-icon>
+              </div>`}
+        </header>
+        <div id="description">
+          <!-- Provide a description for the alert message -->
+          <slot></slot>
+        </div>
+        ${footer}
+      </div>
+    </section>
+    `;
+  }
+
+  firstUpdated(): void {
+    const icons = this.renderRoot.querySelectorAll('#arrow-icon, #close-button');
+    icons.forEach(icon => {
+      icon.addEventListener('click', () => {
+        // Remove active from all icons of all components
+        document.querySelectorAll('pf-alert').forEach(alert => {
+          const innerIcons = alert.renderRoot?.querySelectorAll('#arrow-icon, #close-button');
+          innerIcons?.forEach(i => i.classList.remove('active'));
+        });
+        // Add active to the current icon
+        icon.classList.add('active');
+      });
+    });
+
+    // Click anywhere else returns to normal mode
+    document.addEventListener('click', event => {
```

Comment:
this will add a new document listener for every alert instance, which may cause performance issues. see how popover handles this in https://github.com/patternfly/patternfly-elements/blob/916823a18cc50b9595d91638684665f696dbde7e/elements/pf-popover/pf-popover.ts#L57 and https://github.com/patternfly/patternfly-elements/blob/916823a18cc50b9595d91638684665f696dbde7e/elements/pf-popover/pf-popover.ts#L67-L77 and https://github.com/patternfly/patternfly-elements/blob/916823a18cc50b9595d91638684665f696dbde7e/elements/pf-popover/pf-popover.ts#L353-L358